pkg/controllers/addons_controller.go

```diff
@@ -88,30 +91,42 @@ func (r *AddonsLayerReconciler) processAddonLayer(l layers.Layer) error {
 		return nil
 	}
 
-	if l.IsPruningRequired() {
+	pruneIsRequired, hrs, err := applier.PruneIsRequired(ctx, l)
+	if err != nil {
+		// TODO - we might want to add some sort of error handling here
+		return err
+	} else if pruneIsRequired {
```

Comment:
I'm not a fan of else if here, I think one of the linters we current have suppressed would flag this.
The if err path cause a return (or error recovery) so no need for else if, let's discuss